smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/schema/SchemaGenerator.kt

```diff
@@ -0,0 +1,77 @@
+package software.amazon.smithy.swift.codegen.integration.serde.schema
+
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.swiftmodules.SmithyTypes
+import kotlin.jvm.optionals.getOrNull
+
+class SchemaGenerator(
```

Comment:
This type code-generates a schema for a Smithy model shape.
A couple notes:
- Schemas are generated as computed, global variables, with a naming scheme that ensures each schema has a unique name.  Computed properties and reference types are used to avoid circular reference compile errors on self-referencing Smithy shapes.
- Schema and related types are being marked `Sendable` so that they can be assigned to global vars, but I don't expect them to have any sort of async access or functions.